blivet/nvdimm.py

```diff
@@ -33,21 +34,24 @@
 log_bd_message = lambda level, msg: program_log.info(msg)
 
 
-_REQUESTED_PLUGINS = BlockDev.plugin_specs_from_names(("nvdimm",))
 _HAVE_NVDIMM = False
 
-try:
-    # do not check for dependencies during libblockdev initializtion, do runtime
-    # checks instead
-    BlockDev.switch_init_checks(False)
-    succ_, avail_plugs = BlockDev.try_reinit(require_plugins=_REQUESTED_PLUGINS,
-                                             reload=False,
-                                             log_func=log_bd_message)
-except GLib.GError as err:
-    log.error("Failed to intialize the libblockdev library: %s", err)
-    _HAVE_NVDIMM = False
-else:
-    _HAVE_NVDIMM = "nvdimm" in avail_plugs
+# do not try to load nvdimm plugin on other architectures than x86_64
+# nvdimm (both hardware and plugin) is not availble on other architectures
+if arch.isX86(bits=64):
+    _REQUESTED_PLUGINS = BlockDev.plugin_specs_from_names(("nvdimm",))
+    try:
+        # do not check for dependencies during libblockdev initializtion, do runtime
+        # checks instead
+        BlockDev.switch_init_checks(False)
+        succ_, avail_plugs = BlockDev.try_reinit(require_plugins=_REQUESTED_PLUGINS,
+                                                 reload=False,
+                                                 log_func=log_bd_message)
+    except GLib.GError as err:
+        log.error("Failed to intialize the libblockdev library: %s", err)
+        _HAVE_NVDIMM = False
```

Comment:
This should be redundant since you initialized it above and (I think) this only runs once.